meta-resin-common/recipes-support/resin-init/resin-init-flasher/resin-init-flasher

```diff
@@ -97,6 +97,7 @@ fi
 
 # Flash Resin image on internal device
 inform "Flash internal device... will take around 5 minutes... "
+INTERNAL_DEVICE_KERNEL="${INTERNAL_DEVICE_KERNEL//$(findmnt --noheadings --canonicalize --output SOURCE / | xargs lsblk -no pkname)/}"
```

Comment:
This is quite confusing and looks weak too. I would consider changing to defining a variable for current root device and after that, in the next for, add a check and a message that we ignore that value. It looks cleaner to me.